uber/model_checks.py

```diff
@@ -197,14 +249,17 @@ def job_conflicts(job):
                 return 'You cannot change this job to this time, because {} is already working a shift then'.format(shift.attendee.full_name)
 
 
-cashformpoints_amount = money_amount
-
+@validation.OldMPointExchange
 def oldmpointexchange_numbers(mpe):
     if not str(mpe.amount).isdigit():
         return 'MPoints must be a positive integer'
 
-sale_required = [('what',"What's being sold")]
-def sale_amounts(sale):
+
+Sale.required = [('what', "What's being sold")]
+
+
+@validation.Sale
+def cash_and_mpoints(sale):
     if not str(sale.cash).isdigit() or int(sale.cash) < 0:
         return 'Cash must be a positive integer'
     if not str(sale.mpoints).isdigit() or int(sale.mpoints) < 0:
```

Comment:
Sure thing.  In our SQLAlchemy classes, we have a bunch of foreign keys, with ``relationship()`` fields declared.  We used to declare them like this:

```python
class Attendee(MagModel):
    ...

class Shift(MagModel):
    attendee_id = Column(UUID, ForeignKey('attendee.id', ondelete='cascade'))
    attendee    = relationship(Attendee, backref='shifts')
```

That way if we have a Shift object, we could say ``shift.attendee.full_name`` and whatnot and it would work just like you'd expect.  Ditto if you have an Attendee object and want to say ``attendee.shifts[0]`` or something.

We're still doing all of that, except that we're now declaring the ``relationship`` on the parent instead of the child, e.g.

```python
class Attendee(MagModel):
    shifts = relationship('Shift', backref='attendee')

class Shift(MagModel):
    attendee_id = Column(UUID, ForeignKey('attendee.id', ondelete='cascade'))
```

This does exactly the same thing, except that all of the code examples in SQLAlchemy's documentation do it the latter way.  Last year we did it the former way, and there were a couple of bugs in how we declared things, and the fact that our code looked different from all of the examples I found online made it trickier to debug.  So I just changed the way we declare our relationships to match how everyone else is doing it, i.e. parent-to-child instead of child-to-parent.  That's what I meant by "directionality of our ``relationship`` delcarations", which is retrospect must have been super-unclear :)